lib/mdns_lite/vintage_net_monitor.ex

```diff
@@ -1,108 +1,110 @@
-defmodule MdnsLite.VintageNetMonitor do
-  use GenServer
-  @moduledoc false
-
-  alias MdnsLite.{Responder, ResponderSupervisor}
-
-  @addresses_topic ["interface", :_, "addresses"]
-
-  defmodule State do
+if Code.ensure_loaded?(VintageNet) do
```

Comment:
When a client of mdns_lite does not use the optional VintageNetMonitor, there are warnings like this:
```
warning: VintageNet.match/1 is undefined (module VintageNet is not available or is yet to be defined)
  lib/mdns_lite/vintage_net_monitor.ex:42: MdnsLite.VintageNetMonitor.handle_continue/2
```
This check gets rid of the warnings for the optional code.